05_swap.md

```diff
@@ -125,38 +125,38 @@ Do them in a general way.
 If they're at least relatively efficient, that is, they are not slower than specific things 
 written in the language, then let us talk.
 If you cannot do it, let us stick with C++.
-I'm just explaining the reasoning behind my my choice of C++.
+I'm just explaining the reasoning behind my choice of C++.
 
 ## Swap
 
 Let us look at these three programs.
-Why they are important? 
+Why are they important? 
 Why is swap important? What does it deal with?
 Apparently it's not self-evident.
-Once upon a time I was talking to a very famous program,
+Once upon a time I was talking to a very famous programmer,
 supposedly the best programmer A9 ever had.
 I told him about these three things and  he looks at me
 and said, "I never had to use swap in my life".
 I don't know... I was very impressed because
 you swap for sorting,
 for reversing the sequence,
 for rotating the sequence,
-for all kind of operations.
-Basically, if you do something with a
-sequence you swap.
- so it is very important practically.
-But, it also happens to be very important
-theoretically because a long time ago when people were starting group theory[^group-theory]
-They discovered that any permutation of a sequence could be generated out of swap[^permutation].
+for all kinds of operations.
+Basically if you do something with a
+sequence, you swap.
+So it is very important practically.
+But it also happens to be very important
+theoretically, because a long time ago when people were starting group theory[^group-theory]
+they discovered that any permutation of a sequence could be generated out of swap[^permutation].
 Swap is the most primitive operation.
-the reason is sequence, and any other
-permutation, can be constructed out of swap.
+The reason is sequence. And any other
```

Comment:
This doesn't really scan properly. It could probably do with a little more attention. I think he's trying to say that swap is important because every time you do anything with any sequence you probably need to use swap. And that any permutation of a sequence can be generated using swaps, which shows its power in general.

Reply:
Probably an em-dash would be more appropriate than a full-stop here, but that's not really in keeping with the general style so I shied away from suggesting that.